tests/integration_tests/utils.py

```diff
@@ -164,12 +172,14 @@ def set_feature(**kwargs):
         'max_len': 5,
         'embedding_size': 5
     }
+    feature[ID] = feature[NAME]
+    feature[HASH] = compute_feature_hash(feature)
     feature.update(kwargs)
     return feature
 
 
 def sequence_feature(**kwargs):
-    seq_feature = {
+    feature = {
```

Comment:
To clean this up, I think we should introduce a class called `FeatureDict` that derives from `dict`.  When you initialize it, it should pre-populate the correct ID and HASH values.  This also allows us to add more convenience methods to the `features` in the future.  It can also raise errors if certain fields like NAME are not set.

A better name might be `FeatureMetadata`.

Reply:
This is a good idea